OpenRA.Game/Traits/World/Shroud.cs

```diff
@@ -129,6 +125,11 @@ void AddVisibility(Actor a, ref CPos[] visible)
 			foreach (var c in visible)
 			{
 				var uv = c.ToMPos(map);
+
+				// Force cells outside the visible bounds invisible
+				if (!map.Contains(uv))
+					continue;
```

Comment:
This is going to break the corresponding ```RemoveVisibility``` call since it will remove vis on cells that you're filtering out here.

Maybe expose the ```allowOutsideBounds``` parameter in the ```FindVisibleTiles``` method?